src/lib/EventPublisher.js

```diff
@@ -140,25 +151,30 @@ class EventPublisher extends EventEmitter {
           timeoutHandler();
         }
       }, subscriptionTimeout);
-
-      this.once(eventNamePrefix, (): void => clearTimeout(timeout));
+      this.once(eventNamePrefix, () => {
+        clearTimeout(timeout);
+      });
     }
 
     if (once) {
       this.once(eventNamePrefix, listener);
+      this.once(eventNamePrefix, () => {
+        this._subscriptionsByID.delete(subscriptionID);
+      });
     } else {
       this.on(eventNamePrefix, listener);
     }
-
     return subscriptionID;
   };
 
   unsubscribe = (subscriptionID: string) => {
-    const { eventNamePrefix, listener } = nullthrows(
-      this._subscriptionsByID.get(subscriptionID),
+    const subscription: Subscription = this._subscriptionsByID.get(
+      subscriptionID,
     );
-
-    this.removeListener(eventNamePrefix, listener);
+    if (!subscription) {
+      return;
```

Comment:
1) you shouldn't explicitly cast `subscription: Subscription` Flow is smart enough to understand the type from the map.
2) Your casting is wrong because it's `?Subscription`